bunnet/migrations/runner.py

```diff
@@ -0,0 +1,203 @@
+                        allow_index_dropping=allow_index_dropping
+                    )
+                    migration_node = migration_node.prev_migration
+                    if migration_node is None:
+                        break
+            else:
+                logger.info(f"Running {mode.distance} migrations backward")
+                for i in range(mode.distance):
+                    migration_node.run_backward(
+                        allow_index_dropping=allow_index_dropping
+                    )
+                    migration_node = migration_node.prev_migration
+                    if migration_node is None:
+                        break
+
+    def run_forward(self, allow_index_dropping):
+        if self.forward_class is not None:
+            self.run_migration_class(
+                self.forward_class, allow_index_dropping=allow_index_dropping
+            )
+        self.update_current_migration()
+
+    def run_backward(self, allow_index_dropping):
+        if self.backward_class is not None:
+            self.run_migration_class(
+                self.backward_class, allow_index_dropping=allow_index_dropping
+            )
+        if self.prev_migration is not None:
+            self.prev_migration.update_current_migration()
+        else:
+            self.clean_current_migration()
+
+    def run_migration_class(self, cls: Type, allow_index_dropping: bool):
+        """
+        Run Backward or Forward migration class
+
+        :param cls:
+        :param allow_index_dropping: if index dropping is allowed
+        :return:
+        """
+        migrations = [
+            getattr(cls, migration)
+            for migration in dir(cls)
+            if isinstance(getattr(cls, migration), BaseMigrationController)
+        ]
+
+        client = DBHandler.get_cli()
+        db = DBHandler.get_db()
+        if client is None:
+            raise RuntimeError("client must not be None")
+        with client.start_session() as s:
+            with s.start_transaction():
+                for migration in migrations:
+                    for model in migration.models:
+                        init_bunnet(
+                            database=db,
+                            document_models=[model],  # type: ignore
+                            allow_index_dropping=allow_index_dropping,
+                        )  # TODO this is slow
+                    logger.info(
+                        f"Running migration {migration.function.__name__} "
+                        f"from module {self.name}"
+                    )
+                    migration.run(session=s)
+
+    @classmethod
+    def build(cls, path: Path):
+        """
+        Build the migrations linked list
+
+        :param path: Relative path to the migrations directory
+        :return:
+        """
+        logger.info("Building migration list")
+        names = []
+        for modulepath in path.glob("*.py"):
+            names.append(modulepath.name)
```

Comment:
Hi @roman-right, could` __init__.py` be excluded here?

The problem we're facing is that only the initial scripts are run (and any later added migration files are ignored in further runs).
This is because the first migration creates following records in `migrations_log` (`_id` and `ts` omitted):  
```
name: 20221103203943_migration_test.py, is_current: false
name: 20221103204129_migration_test_2.py, is_current: false
name: __init__.py, is_current: true
```
and newly added migration files (which names start with a timestamp) never appear after `__init__.py` when sorted alphabetically.

We wish to have `__init__.py` in the package along with the migration files in order to be able to import the migration files in unit tests.

Reply:
Hi @PetrF0X ,
Good point! I'll exclude __init__ file there. Thank you.